src/rgw/rgw_op.cc

```diff
@@ -3654,6 +3654,197 @@ void RGWDeleteMultiObj::execute()
 
 }
 
+bool RGWBulkDelete::Deleter::verify_permission(RGWBucketInfo& binfo,
+                                               map<string, bufferlist>& battrs,
+                                               rgw_obj& obj,
+                                               ACLOwner& bucket_owner /* out */)
+{
+  int ret = 0;
+
+  RGWAccessControlPolicy bacl(store->ctx());
+  rgw_obj_key no_obj;
+  ret = read_policy(store, s, binfo, battrs, &bacl, binfo.bucket, no_obj);
+  if (ret < 0) {
+    return false;
+  }
+
+  RGWAccessControlPolicy oacl(s->cct);
+  ret = read_policy(store, s, binfo, battrs, &oacl, binfo.bucket, s->object);
+  if (ret < 0) {
+    return false;
+  }
+
+  bucket_owner = bacl.get_owner();
+
+  return verify_object_permission(s, &bacl, &oacl, RGW_PERM_WRITE);
```

Comment:
@rzarzynski do we need write permissions to the object? In RGWDeleteObj::verify_permission() we only check that the bucket has write permissions.